backend/src/tasks/lookingGlass.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  Organization,
+  Domain,
+  Vulnerability,
+  connectToDatabase
+} from '../models';
+import { CommandOptions } from './ecs-client';
+import got from 'got';
+import { plainToClass } from 'class-transformer';
+import saveVulnerabilitiesToDb from './helpers/saveVulnerabilitiesToDb';
+import { exit } from 'process';
+import saveDomainsToDb from './helpers/saveDomainsToDb';
+
+interface LGCollectionsResponse {
+  children: any[];
+  createdAt: string;
+  createdBy: string;
+  id: string;
+  name: string;
+  ticScore: number;
+  updatedAt: string;
+  updatedBy: string;
+}
+
+interface LGThreatResponse {
+  totalHits: number;
+  results: {
+    firstSeen: number;
+    lastSeen: number;
+    sources: string[];
+    right: {
+      ticClassificationScores: number[];
+      threatId: string;
+      classifications: string[];
+      ticScore: number;
+      ticCriticality: number;
+      ticSourceScore: number;
+      ticObsInfluence: number;
+      ref: {
+        type: string;
+        id: string;
+      };
+      threatCategories: string[];
+      name: string;
+      type: string;
+      threatName: string;
+    };
+    left: {
+      collectionIds: string[];
+      classifications: string[];
+      ticScore: number;
+      owners: string[];
+      ref: {
+        type: string;
+        id: number;
+      };
+      name: string;
+      type: string;
+      threatNames: string[];
+      locations: {
+        city: string;
+        country: string;
+        region: string;
+        geoPoint: {
+          long: number;
+          lat: number;
+        };
+        countryName: string;
+        sources: string[];
+        country2Digit: string;
+        lastSeen: number;
+      }[];
+      ipv4: number;
+      asns: number[];
+      threatIds: string[];
+      cidrv4s: string[];
+    };
+    ref: {
+      type: string;
+      right: {
+        type: string;
+        id: string;
+      };
+      left: {
+        type: string;
+        id: string;
+      };
+    };
+  }[];
+}
+
+async function getCollectionForCurrentWorkspace() {
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    process.env.LG_WORKSPACE_NAME +
+    '/collections';
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function validateIPAddress(ipAddress) {
+  return /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipAddress);
+}
+
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    period: 'all',
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
```

Comment:
Is this limit OK? Is it possible to exceed this limit?

Reply:
I'm looking into it now, We only need the last months worth of Vulns or Mals, so I don't think there will be over 100,000 in the last month, but I need to verify that they are returned in chronological order

Reply:
right now the closest is 13000 so i dont think we will hit it anytime soon, and that time we will have most likely stopped using lookingGlass